lib/_stream_writable.js

```diff
@@ -473,8 +486,14 @@ function finishMaybe(stream, state) {
   if (need) {
     if (state.pendingcb === 0) {
       prefinish(stream, state);
-      state.finished = true;
-      stream.emit('finish');
+      state.flushed = true;
+      state._end(stream, function(err) {
+        if (err)
+          stream.emit('error', err);
```

Comment:
I don't think so. `state.flushed` is already true so this part of code won't be re-entered.